### ISSUE TITLE
✨ Improvement:  login flow removal

### ISSUE THREAD

**Issue by VineeTagarwaL-code** [code100x/job-board#201]
### 📜 Description

Objective:
The goal is to streamline the user experience by removing the login and signup process for general users, making the job portal open to all without any barriers. This will simplify the user flow, similar to platforms like RemoteJobs, where users can easily access job listings without any hassle.

Tasks:

    Remove Authentication for General Users:
        Eliminate the login/signup process for users accessing the job portal.
        Ensure that all job listings and related content are publicly accessible without requiring user authentication.

    Preserve Admin Authentication:
        Maintain the authentication flow for the admin role.
        Admins should still be able to access their specific protected routes to manage jobs in the database.
        Focus on general user access in this task, leaving admin role management and portal development for a later phase.

Next Steps:
After this task is completed, we will focus on enhancing the admin role functionalities and protecting specific routes as needed.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/code100x/job-board/blob/main/CONTRIBUTING.md)

**Comment by OmkarK97** [comment]
@VineeTagarwaL-code would like to work on this

**Comment by Ameerjafar** [comment]
@VineeTagarwaL-code can you assign this issue to work

**Comment by Ameerjafar** [comment]
@VineeTagarwaL-code without login/signup we can able to do job listings and related contents in the current code we don't need to login to get the content (general users)

**Comment by VineeTagarwaL-code** [comment]
hey @Ameerjafar yes assigning you for this, leave your doubts below before starting to work on this

1. yeah users can see and apply on jobs that are present without any flow

**Comment by Ameerjafar** [comment]
What about the idea  made by @thrishank? 
@VineeTagarwaL-code

**Comment by Ameerjafar** [comment]
@VineeTagarwaL-code the issue was already done, general users have no restrictions in any way. The administrator must check in before he can access the portal to perform any administrative tasks.

**Comment by Ameerjafar** [comment]
@VineeTagarwaL-code

**Comment by VineeTagarwaL-code** [comment]
i want to remove the signin feature wholely, no need of that for now 


@thrishank yeah but we are not making something like linkedin, we are simply job posting portal for now ... companies can come and have there openings and you can apply to the lnk provided by them its as easy as possible

**Comment by Ameerjafar** [comment]
@VineeTagarwaL-code then how do we know the user is general or admin if we remove the entire signin process because any one can manage the jobs (hardcoded the admin signin email)

**Comment by anuragnegi000** [comment]
> @VineeTagarwaL-code then how do we know the user is general or admin if we remove the entire signin process because any one can manage the jobs (hardcoded the admin signin email)

 we can introduce a special route exclusively for admins. This route would be shared only by the superadmin.

How would this work?
Admins would access a login page through this special route. Upon attempting to log in, the superadmin would receive an email notification indicating that a person is requesting admin access. Once the superadmin grants access, the admin can log in and post jobs.

Why would the superadmin receive an email, even if they shared the route themselves?
This is to confirm that the person attempting to log in is indeed the intended recipient of the special route.

**Comment by anuragnegi000** [comment]
> > @VineeTagarwaL-code then how do we know the user is general or admin if we remove the entire signin process because any one can manage the jobs (hardcoded the admin signin email)
> 
> we can introduce a special route exclusively for admins. This route would be shared only by the superadmin.
> 
> How would this work? Admins would access a login page through this special route. Upon attempting to log in, the superadmin would receive an email notification indicating that a person is requesting admin access. Once the superadmin grants access, the admin can log in and post jobs.
> 
> Why would the superadmin receive an email, even if they shared the route themselves? This is to confirm that the person attempting to log in is indeed the intended recipient of the special route.

@VineeTagarwaL-code i would like to contribute into this if anyone else is not

**Comment by VineeTagarwaL-code** [comment]
@anuragnegi000 go for it

**Comment by VineeTagarwaL-code** [comment]
> @VineeTagarwaL-code then how do we know the user is general or admin if we remove the entire signin process because any one can manage the jobs (hardcoded the admin signin email)

we will hard code admin, simple
